Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -758,19 +807,58 @@ void doEsirkepovDepositionShapeN (const GetParticlePosition<PIdx>& GetPosition,
 #if !defined(WARPX_DIM_1D_Z)
             double sx_new[depos_order + 3] = {0.};
             double sx_old[depos_order + 3] = {0.};
-            const int i_new = compute_shape_factor(sx_new+1, x_new);
+            const int i_new = compute_shape_factor(sx_new+1, x_new );
             const int i_old = compute_shifted_shape_factor(sx_old, x_old, i_new);
+            // If particle is close to the embedded boundary, recompute deposition with order 1 shape
+            if constexpr (reduce_shape_control == has_reduced_shape) {
+                if (reduce_shape_new) {
+                    for (int i=0; i<depos_order+3; i++) sx_new[i] = 0.; // Erase previous deposition
+                    compute_shifted_shape_factor_order1( sx_new+depos_order/2, x_new, i_new+depos_order/2 ); // Redeposit with order 1
+                }
+                if (reduce_shape_old) {
+                    for (int i=0; i<depos_order+3; i++) sx_old[i] = 0.; // Erase previous deposition
+                    compute_shifted_shape_factor_order1( sx_old+depos_order/2, x_old, i_new+depos_order/2 ); // Redeposit with order 1
+                }
+                // Note: depos_order/2 in the above code corresponds to the shift between the index of the lowest point
```

Comment:
Thanks for the comment, very helpful to parse this.
If it helps to name this, you could also write this in an `const int` before the function call.